### DISCUSSION TITLE
CSV comparison failing.  MRE

### DISCUSSION THREAD

**Discussion by eogoodwin** [WinMerge/winmerge#1471] [Q&A] [answered]
Hi,

I'm hoping someone can illuminate for me, what settings I might have fiddled with, to prevent winMerge from finding the differences between these files.

I've trimmed the files down to a minimum reproducible, but it might depend on my config.

[WinMerge.txt](https://github.com/WinMerge/winmerge/files/9465776/WinMerge.txt)

![image](https://user-images.githubusercontent.com/34554459/187803587-5ab65d07-42f8-44ef-8dfe-5fa6510c5259.png)

[cf1.csv](https://github.com/WinMerge/winmerge/files/9465763/cf1.csv)
[cf2.csv](https://github.com/WinMerge/winmerge/files/9465766/cf2.csv)

Have I got it ignoring quotated text or something? But, the single capital characters in the last columns, ...   i dont get it :-)

Cheers,

Eric

**Comment by sdottaka** [comment, marked as answer]
It appears that the following line filter is enabled. Try disabling this.

![image](https://user-images.githubusercontent.com/98126/187913930-14e2f967-e90d-4c79-aef0-d53263f000fe.png)

**Reply by eogoodwin** [reply]
Indeed! That was it. Thank you.
I guess I had set that some time ago, and forgotten that it was active.  I hadn't found it in the settings when I was looking for anything that might be preventing those differences being seen. Thank you for finding it for me :-)
You might be able to help me further - I thought I remembered using a function that would ignore specific differences, such as "ignore a '2019' changing to a '2020'", which might be what I was trying to set up with that line filter.  Do you know of a mechanism to implement that sort of exception?  Is that a substitution filter perhaps?

**Reply by sdottaka** [reply]
As you can imagine, you can substitute 2019 for 2020 with the following Substitution filters and compare.

![image](https://user-images.githubusercontent.com/98126/188166009-7e154349-23ce-43dd-bae8-d3a2fc852399.png)

**Comment by eogoodwin** [comment]
Thanks for your help again.  Is the find / replace perhaps confused by punctuation?  I've tried a number of substitution filters to ignore the "riversNI" versus "riversSI" difference, but still get that flagged as a difference.

![image](https://user-images.githubusercontent.com/34554459/205525969-3b2f2890-982c-4865-be61-4878be6ccaae.png)

I've tried filters like those shown, separately so that they don't conflict, or together "just in case", but always still get the differences flagged.  Unfortunately the "substitution filter" section of my help file still shows "TBD" 🙂

Is there a better syntax for what I'm trying to do here?

Thanks,  Eric.

**Reply by sdottaka** [reply]
I have created and compared files similar to the example you have shown, and the substitution filters seem to work as follows.
Do you see the same results in your environment when comparing the files I have attached?
Note that if you apply the substitution filters and find that some words in a line are substituted and match, but there are other differences in the line, the substituted differences are not ignored.

![image](https://user-images.githubusercontent.com/98126/205644831-b8efa4dc-4a84-45c5-b8f4-69ed097b089d.png)
[#1471.zip](https://github.com/WinMerge/winmerge/files/10154385/1471.zip)

**Comment by eogoodwin** [comment]
Thanks Sawanaka, for your patience and support.  You're right, that in isolation, those file fragments are evaluated as identical.  You're right that there were other differences in some lines, sometimes there were "NI" vs "SI" differences in file paths, that werent caught by my substitution filters of "riversNI" and "riversSI".
But I still see some other differences that I can explain.  Differences that are detected when lines are in their original context, but not when I extract them from the files and isolate them.

![image](https://user-images.githubusercontent.com/34554459/205730276-c4c794cd-d6c9-4691-8b0d-bba9c85de07f.png)

I'm showing here two pairs of files.  The upper one contains the entire scripts I'm comparing, the lower one contains only ten lines that I copied and pasted.  Differences are detected in context, but not in isolation.  I cant see that I have file filters or line filters in effect, so these differences remain a mystery to me.  No doubt it's something else that I don't yet understand.  You've been very helpful to me so far, thank you.

In one case, a line pair that was identified as different, was identified as identical when I made sure to match the number of newLines following it.  There was no effective difference within the line pair, only a difference in the number of empty newLines following it, caused it to be flagged as a difference.  That doesnt look to be the case in this example.

Eric

**Reply by sdottaka** [reply]
I couldn't tell from the pictures alone. Could you please attach the file so that I can reproduce it here?

Perhaps this is due to the restrictions that Substitution filters are applied per Diff block instead of per line

What kind of results can you get if you try the PrediffLineFilter plugin described in the URL below?

https://github.com/WinMerge/winmerge/issues/796#issuecomment-979114453
